workspace_integration_test.go

```diff
@@ -217,6 +217,26 @@ func TestWorkspacesList(t *testing.T) {
 		assert.NotEmpty(t, wTest.ID)
 		assert.Equal(t, 0, wl.TotalCount)
 	})
+
+	t.Run("when filtering using project id", func(t *testing.T) {
+		p, pTestCleanup := createProject(t, client, orgTest)
+		defer pTestCleanup()
+
+		// create a workspace with project
+		w, wTestCleanup := createWorkspaceWithOptions(t, client, orgTest, WorkspaceCreateOptions{
+			Name:    String(randomString(t)),
+			Project: p,
+		})
+		defer wTestCleanup()
+		// Request a page number which is out of range. The result should
+		// be successful, but return no results if the paging options are
+		// properly passed along.
+		wl, err := client.Workspaces.List(ctx, orgTest.Name, &WorkspaceListOptions{
+			ProjectID: p.ID,
+		})
+		require.NoError(t, err)
+		assert.Contains(t, wl.Items, w)
```

Comment:
Hmm, to properly test the filter, I think we also need a test for _exclusion_ from the list. 

I'd be satisfied with either: 
- Create a workspace that lands in the default project, and confirm that it's excluded from the filtered list.
- Create another project and do a second search, confirming that our workspace is excluded when searching that other project.

I think we'd only need one of those, so whichever's comfier.

Reply:
Added one where workspaces are added to the default project automatically, and the new project is empty.